clusterloader2/testing/storage/pod-startup/ephemeral-volumes/config.yaml

```diff
@@ -0,0 +1,97 @@
+#Constants
+{{$NODES_PER_NAMESPACE := DefaultParam .NODES_PER_NAMESPACE 1}}
+{{$PODS_PER_NODE := DefaultParam .PODS_PER_NODE 1}}
+{{$LOAD_TEST_THROUGHPUT := DefaultParam .LOAD_TEST_THROUGHPUT 10}}
+
+{{$VOLUMES_PER_POD := 4}}
+{{$VOLUME_TYPE := DefaultParam .VOLUME_TYPE "EmptyDir"}}
+{{$PROVISION_VOLUME := DefaultParam .PROVISION_VOLUME false}}
+
+{{$APP_NAME := "pod-load"}}
+{{$GROUP := "pod-with-ephemeral-volume-startup-latency"}}
+
+#Variables
+# TODO think about this!
+{{$podStartupTimeout := $VOLUMES_PER_POD}}
+{{$namespaces := DivideInt .Nodes $NODES_PER_NAMESPACE}}
+{{$podsPerNamespace := MultiplyInt $NODES_PER_NAMESPACE $PODS_PER_NODE}}
+{{$totalPods := MultiplyInt $namespaces $NODES_PER_NAMESPACE $PODS_PER_NODE}}
+{{$volumesPerNamespace := MultiplyInt $podsPerNamespace $VOLUMES_PER_POD}}
+
+name: pod-with-ephemeral-volume-startup-latency
+automanagedNamespaces: {{$namespaces}}
+tuningSets:
+# TODO different options?
+- name: UniformQPS
+  qpsLoad:
+    qps: {{$LOAD_TEST_THROUGHPUT}}
+steps:
+- measurements:
+  - Identifier: APIResponsiveness
+    Method: APIResponsiveness
+    Params:
+      action: reset
+{{ if $PROVISION_VOLUME }}
+# Provision volumes
+- phases:
+  - namespaceRange:
+      min: 1
+      max: {{$namespaces}}
+    replicasPerNamespace: {{$volumesPerNamespace}}
+    tuningSet: UniformQPS
+    objectBundle:
+    - basename: vol-{{$APP_NAME}}
+      objectTemplatePath: {{$VOLUME_TYPE}}.yaml
+{{ end }}
+# Start measurements
```

Comment:
I vote for moving this up and doing measurement operations together.

Reply:
I'm not sure about this, because KEP says that [`assuming that the volumes have already been provisioned`](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190204-scale-testing.md#pod-startup).

If I move this up, then measurement is started before volume provisioning.

Reply:
I think we want to separate out provisioning time measurements from pod startup measurements because provisioning is generally a one-time bootstrapping operation (for an application).

Reply:
> If I move this up, then measurement is started before volume provisioning.

yes - but the measurement only measures pod startup time. the fact that you create some other objects doesn't affect that at all.

We want to separate out provisioning time - but that means creating those volumes before we create a pod - which is happening in this test no matter when you start the measurement.

Reply:
When does pod startup measurement begin?  For ephemeral volumes, kubelet cannot mount API volumes until the API object is created, and containers cannot start until all volumes are mounted. For PVCs, scheduler cannot schedule the pod until the pvs are provisioned and bound to the PVC.

Reply:
The "measurement start" action in the config is basically "start watching for pod creations and for those measure pod startup time".
Measuring pod startup time for individual pod starts when that particular pod is created, which in case of this config are created after all volumes already exist.